src/twisted/trial/test/matchers.py

```diff
@@ -93,3 +97,46 @@ def getContent(p: IFilePath) -> str:
             return f.read().decode(encoding)
 
     return after(getContent, m)
+
+
+class _MatchInterfaceProvider(BaseMatcher[object]):
```

Comment:
I don't know what to say about these matchers :)

The assertion helper is needed.

I have something similar using old-school xunit assertion style :)

```python
    def assertProvides(self, interface, obj):
        self.assertTrue(
            interface.providedBy(obj),
            'Object %s does not provided interface %s.' % (obj, interface))
        verifyObject(interface, obj)

    def assertNotProvides(self, interface, obj):
        self.assertFalse(
            interface.providedBy(obj),
            'Object %s does not provided interface %s.' % (obj, interface))
```

https://github.com/chevah/compat/blob/master/src/chevah_compat/testing/assertion.py#L219-L228


Maybe, I am just getting old, but I find these hamcrest part  too complicated :)

I know that old-style xunit is based on inheritance , but we are already writing tests in xunit style